### ISSUE TITLE
Automatically remap commands in Select Mode

### ISSUE THREAD

**Issue by AntonioLucibello** [helix-editor/helix#2125]
Select mode's default keymap is currently hardcoded to mirror normal mode's default keymap, and does not update to reflect changes when remapping normal mode.

This is somewhat counterintuitive to the user, since select mode is presented as just a modifier that makes movement keys extend selections (to the point that select mode commands aren't shown in the docs).
In other words, if I remapped hjkl to neio (colemak), I would expect SEL to behave accordingly.

Furthermore, since SEL copies all of NOR's keybinds, even those that aren't actively repurposed by SEL, pressing **any** key in SEL will actually use the default keybind rather than the remapped one. 
For example, if one remapped `undo` to `y` and `yank` to `u`, pressing `u` in SEL to yank a newly extended selection will actually call undo, even though from a user's perspective SEL should have nothing to do with either `yank` or `undo`.

**Comment by perlinm** [comment]
I agree.  My `config.toml` file has a lot of material duplicated under `[keys.normal]` and `[keys.select]`, which is annoying to maintain (especially for non-qwerty users, and/or users that play around with keybindings a lot :slightly_smiling_face:).

Two possible fixes:
- SEL inherits NOR keybindings, with `move_*` commands changed to their `extend_*` versions.  This sounds like what @AntonioLucibello is asking for, and makes the most sense to me.  This solution should also support overriding keybindings in SEL.
- Supporting something like a `[keys.normal_and_select]` section in `config.toml`.  This is probably much easier to implement.

**Comment by chtenb** [comment]
Related issue: https://github.com/helix-editor/helix/issues/1513
